File: recursion and backtracking/recursion_in_an_array/keypad.py
# Errors out
codes = [".;:","abc","def" ,"ghi", "jkl", "mno", "pqrs", "tu","vwx","yz"]

def keypad(numbers):
    if len(numbers) ==0:
        bres = list()
        bres.append("")
        return bres
#    573  -- faiith 73 will bring reesults
    char = numbers[0] #5
    ros = numbers[1:] #73

#     Faith
    rres = keypad(ros)

    mres = list()
    for i in range(0, len(codes[int(char)])):
        code = char[i]
        for rstr in rres:
            mres.append(code+rstr)

    return mres


def main():

    numbers = input()
    result = keypad(numbers)
    print(result)

main()

File: recursion and backtracking/recursion_in_an_array/keypad.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner scn = new Scanner(System.in);
        String str = scn.nextInt();
        ArrayList<String> words = getKPC(str);
        System.out.println(words);

    }

    static String[] codes = {".,", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tu", "vwz", "yz"};
    public static ArrayList<String> getKPC(String str) {
        if(str.length()==0){
            ArrayList<String> bres= new ArrayList<>();
            bres.add("");
            return bres;
        }


        char ch = str.charAt(0);
        String ros = str.substring(1);

        ArrayList<String> rres = getKPC(ros);
        ArrayList<String> mres = new ArrayList<>();

        String codesforch = codes[ch];
        for(int i =0; i<codesforch.length();i++){
            char chcode =codesforch.charAt(i);
            for(String rstr:rres){
                mres.add(chcode +rstr);
            }
        }
        return mres;
    }

}